diffus/src/lcs.rs

```diff
@@ -19,12 +22,20 @@ impl<T: Eq> Lcs<T> {
 
         let mut storage = vec![0; width * height];
 
+        let a = |i: usize, j: usize| {
+            let result = j * width + i;
+
+            result
+        };
+
         for (i, x) in x.enumerate() {
             for (j, y) in y().enumerate() {
-                storage[(i + 1) * width + (j + 1)] = if x == y {
-                    storage[i * width + j] + 1
+                let i = i + 1;
```

Comment:
I find this more confusing. First we add 1 here then we subtract one below. The `a` closure (or `TwoDVec`) helps, but the mutation of the `i` and `j` aliases? Inlining feels easier to follow

Reply:
not when you are looking at the wiki reference, but besides that, yes

Reply:
or, it makes more sense to me to actually be setting the position `storage[i, j]` and have that as a reference point
but I don't feel too strongly about it, the next time I'm in this code och har mig I will probably try to rewrite it to something more readable anyways :) (this will probably never happen)